iota-execution/latest/iota-adapter/src/execution_engine.rs

```diff
@@ -239,6 +242,226 @@ mod checked {
+            ExecutionStatus::new_failure(status, command)
+        } else {
+            ExecutionStatus::Success
+        };
+
+        #[skip_checked_arithmetic]
+        trace!(
+            tx_digest = ?authenticated_transaction_digest,
+            computation_gas_cost,
+            "Finished authenticator execution of transaction with status {:?}",
+            status
+        );
+
+        execution_result.map(|_| computation_gas_cost)
+    }
+
+    /// Executes an authentication move call by processing the specified
+    /// `ProgrammableTransaction`, running the main execution logic.
+    /// Similarly to `execute_transaction`, this function handles certain error
+    /// conditions such as denied certificate, deleted input objects failed
+    /// consistency checks.
+    ///
+    /// Gas costs are managed through the `GasCharger` argument and charged only
+    /// for the authenticator move call execution.
+    ///
+    /// Returns the move authenticator computation gas cost without bucketing
+    /// and the execution result.
+    #[instrument(name = "auth_execute", level = "debug", skip_all)]
+    fn execute_authenticator_move_call<Mode: ExecutionMode>(
+        temporary_store: &mut TemporaryStore<'_>,
+        authenticator_move_call: ProgrammableTransaction,
+        mut gas_charger: GasCharger,
+        tx_ctx: &mut TxContext,
+        move_vm: &Arc<MoveVM>,
+        protocol_config: &ProtocolConfig,
+        metrics: Arc<LimitsMetrics>,
+        deny_cert: bool,
+        contains_deleted_input: bool,
+        cancelled_objects: Option<(Vec<ObjectID>, SequenceNumber)>,
+        trace_builder_opt: &mut Option<MoveTraceBuilder>,
+    ) -> (u64, Result<Mode::ExecutionResults, ExecutionError>) {
+        debug_assert!(
+            gas_charger.no_charges(),
+            "At this point no gas charges must be applied yet"
+        );
+
+        // Charge gas for reading the Move authenticator input objects from the storage.
+        let result = gas_charger.charge_input_objects(temporary_store);
+
+        let result = result.and_then(|()| {
+            run_inputs_checks(
+                protocol_config,
+                deny_cert,
+                contains_deleted_input,
+                cancelled_objects,
+            )?;
+            programmable_transactions::execution::execute::<Mode>(
+                protocol_config,
+                metrics.clone(),
+                move_vm,
+                temporary_store,
+                tx_ctx,
+                &mut gas_charger,
+                authenticator_move_call,
+                trace_builder_opt,
+            )
+            .and_then(|ok_result| {
+                temporary_store.check_move_authenticator_results_consistency()?;
+                Ok(ok_result)
+            })
+        });
+
+        let gas_status = gas_charger.into_gas_status();
+
+        // Return the gas used for execution without bucketing.
+        (gas_status.gas_used() * gas_status.gas_price(), result)
```

Comment:
I think it's worth make it explicit, for clarity:

```suggestion
        // Compute the gas used for the authentication execution without rounding/bucketing.
        let gas_status = gas_charger.into_gas_status();
        let authentication_computation_cost = gas_status.gas_used() * gas_status.gas_price();

        (authentication_computation_cost, result)
```

Reply:
fixed in 7966fab551